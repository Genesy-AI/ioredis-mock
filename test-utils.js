// Runs both a regular string suite and a buffer suite
export function runTwinSuite(commandName, cb) {
  const commands = [`${commandName}Buffer`, commandName]

  return commands.forEach(command =>
    command === commandName
      ? cb(command, (a, b) => a === b)
      : cb(
          command,
          (a, b) => a.equals(Buffer.from(b))
          /*
          // @TODO: detect when it happens so we can remove this
          if (Buffer.isBuffer(a)) return a.equals(Buffer.from(b))

          console.warn('a was expected to be a Buffer:', typeof a, { a, b })
          return a === b
          // */
        )
  )
}

// Util for calling describe.skip if the suite asserts Buffer outputs, which will fail in test:browser.js due to the polyfill
export const browserSafeDescribe = command =>
  process.env.IS_BROWSER && command.endsWith('Buffer')
    ? // eslint-disable-next-line no-undef
      describe.skip
    : // eslint-disable-next-line no-undef
      describe
