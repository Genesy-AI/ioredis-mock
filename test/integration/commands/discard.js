import Redis from 'ioredis'

// eslint-disable-next-line import/no-relative-parent-imports
import { browserSafeDescribe, runTwinSuite } from '../../../test-utils'

runTwinSuite('discard', command => {
  browserSafeDescribe(command)(command, () => {
    it('should discard any batch queue ', async () => {
      expect.assertions(1)
      const redis = new Redis()

      await redis.multi({ pipeline: false })
      await redis.incr('user_next')
      const result = await redis[command]()
      expect(result).toMatchSnapshot()
    })

    it('errors if you discard without starting a pipeline', async () => {
      expect.assertions(1)
      const redis = new Redis()

      try {
        await redis[command]()
      } catch (err) {
        expect(err.message).toMatchSnapshot()
      }
    })
  })
})
