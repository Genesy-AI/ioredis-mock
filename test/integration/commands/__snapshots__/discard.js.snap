// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`discard errors if you discard without starting a pipeline 1`] = `"ERR DISCARD without MULTI"`;

exports[`discard should discard any batch queue  1`] = `"OK"`;

exports[`discardBuffer errors if you discard without starting a pipeline 1`] = `"ERR DISCARD without MULTI"`;

exports[`discardBuffer should discard any batch queue  1`] = `Buffer<4f 4b>`;
